c7n/resources/rds.py

```diff
@@ -281,6 +282,107 @@ def process(self, dbs):
+        version = resource['EngineVersion']
+        versions = client.describe_db_engine_versions(
+            Engine=resource['Engine'],
+            EngineVersion=version)['DBEngineVersions'][0]
+
+        for v in versions['ValidUpgradeTarget']:
+            if v['IsMajorVersionUpgrade']:
+                continue
+
+            if parse_version(v['EngineVersion']) > parse_version(version):
+                version = v['EngineVersion']
+        return version
+
+    def _tag_rds(self, client, resource, version):
+        client.add_tags_to_resource(
+            ResourceName=resource['DBInstanceArn'],
+            Tags=[{
+                'Key': 'engine_target_upgrade',
+                'Value': version}])
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('rds')
+        results = []
+        engines = {}
+        for r in resources:
+            if r['Engine'] not in engines:
+                engines[r['Engine']] = self._list_versions(client, r)
+
+            if r['EngineVersion'] == engines[r['Engine']]:
+                continue
+            if parse_version(
+                    engines[r['Engine']]) > parse_version(r['EngineVersion']):
+                if not r in results:
+                    self._tag_rds(client, r, engines[r['Engine']])
+                    results.append(r)
+        return results
+
+
+@actions.register('auto-upgrade-set')
+class SetAutoUpgradeFlag(BaseAction):
+
+    schema = type_schema('auto-upgrade-set', value={'type': 'boolean'})
+
+    def _set_flag(self, client, resource):
+        client.modify_db_instance(
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'],
+            AutoMinorVersionUpgrade=self.data.get('value', True))
+
+    def process(self, resources):
+        futures = []
+        client = local_session(self.manager.session_factory).client('rds')
+        for r in resources:
+            self._set_flag(client, r)
+
+
+@actions.register('upgrade-minor')
+class UpgradeMinorRDS(BaseAction):
+
+    schema = type_schema('upgrade-minor')
+
+    def _engine_version(self, client, resource):
+        tags = client.list_tags_for_resource(
+            ResourceName=resource['DBInstanceArn'])['TagList']
+        for t in tags:
+            if t['Key'] == 'engine_target_upgrade':
+                return t['Value']
+        return None
+
+    def _upgrade(self, client, resource, version):
+        client.modify_db_instance(
+            DBInstanceIdentifier=resource['DBInstanceIdentifier'],
+            EngineVersion=version,
+            ApplyImmediately=True)
+
+        client.remove_tags_from_resource(
+            ResourceName=resource['DBInstanceArn'],
+            TagKeys=['engine_target_upgrade', 'custodian_upgrade'])
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('rds')
+        for r in resources:
+            version = self._engine_version(client, r)
```

Comment:
version should come from the annotation, and if not present default to latest minor version available, else this is effectively trusting user input as a passthrough to the api call.